From mreimer@sdgsystems.com  Thu Apr 29 12:48:56 2010
From: Matt Reimer <mreimer@sdgsystems.com>
Date: Thu,  1 Apr 2010 13:44:04 -0700
Subject: USB: pxa27x_udc: use four bits to store endpoint addresses
To: linux-usb@vger.kernel.org
Cc: Matt Reimer <mreimer@sdgsystems.com>
Message-ID: <1270154644-13011-1-git-send-email-mreimer@sdgsystems.com>


Endpoint addresses on pxa27x can be programmed as 1-15, but since
only three bits were being used to store the endpoint number it
was possible to overflow.

Signed-off-by: Matt Reimer <mreimer@sdgsystems.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/pxa27x_udc.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/gadget/pxa27x_udc.h
+++ b/drivers/usb/gadget/pxa27x_udc.h
@@ -360,7 +360,7 @@ struct pxa_ep {
 	 * Specific pxa endpoint data, needed for hardware initialization
 	 */
 	unsigned		dir_in:1;
-	unsigned		addr:3;
+	unsigned		addr:4;
 	unsigned		config:2;
 	unsigned		interface:3;
 	unsigned		alternate:3;
